File: app/src/main/java/com/smartnsoft/updatepopupsample/CustomUpdatePopupActivity.java
package com.smartnsoft.updatepopupsample;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.text.Html;
import android.text.TextUtils;
import android.view.View;

import com.bumptech.glide.Glide;
import com.smartnsoft.smartappupdate.AbstractSmartAppUpdateActivity;

/**
 * @author Adrien Vitti
 * @since 2018.01.25
 */

public final class CustomUpdatePopupActivity
    extends AbstractSmartAppUpdateActivity
{

  @Override
  protected void setImage(@Nullable String imageURLFromRemoteConfig)
  {
    if (TextUtils.isEmpty(imageURLFromRemoteConfig) == false)
    {
      image.setVisibility(View.VISIBLE);
      Glide.with(this).load(imageURLFromRemoteConfig).into(image);
    }
  }

  @Override
  protected void setContent(@Nullable String contentFromRemoteConfig)
  {
    if (TextUtils.isEmpty(contentFromRemoteConfig) == false)
    {
      paragraph.setText(Html.fromHtml(contentFromRemoteConfig));
    }
  }

  @Override
  public void sendUpdateInfoDisplayed()
  {

  }

  @Override
  public void sendRecomendedUpdateDisplayed()
  {

  }

  @Override
  public void sendBlockingUpdateDisplayed()
  {

  }

  @Override
  public void sendAskLaterEvent()
  {

  }

  @Override
  public void closeAppWithoutUpdateEvent()
  {

  }

  @Override
  public void sendInformativeActionButtonEvent()
  {

  }

  @Override
  public void sendRecommendedActionButtonEvent()
  {

  }

  @Override
  public void sendBlockingActionButtonEvent()
  {

  }

  @Override
  public int getVersionCode()
  {
    return 0;
  }

  @Override
  public String getDateForAnalytics()
  {
    return null;
  }

  @Override
  public Bundle generateAnalyticsExtraInfos()
  {
    return null;
  }
}

File: app/src/main/java/com/smartnsoft/updatepopupsample/CustomUpdatePopupActivity.kt
package com.smartnsoft.updatepopupsample

import android.os.Build
import android.os.Bundle
import android.text.Html
import android.view.View
import com.bumptech.glide.Glide
import com.smartnsoft.smartappupdate.AbstractSmartAppUpdateActivity

/**
 * @author Adrien Vitti
 * @since 2018.01.25
 */

class CustomUpdatePopupActivity : AbstractSmartAppUpdateActivity()
{

  override fun setImage(imageURLFromRemoteConfig: String?)
  {
    if (imageURLFromRemoteConfig.isNullOrBlank())
    {
      return
    }
    image.visibility = View.VISIBLE
    Glide.with(this).load(imageURLFromRemoteConfig).into(image)
  }

  override fun setContent(contentFromRemoteConfig: String?)
  {
    if (contentFromRemoteConfig.isNullOrBlank())
    {
      return
    }
    paragraph.text = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
    {
      Html.fromHtml(contentFromRemoteConfig, Html.FROM_HTML_MODE_LEGACY)
    }
    else
    {
      @Suppress("DEPRECATION")
      Html.fromHtml(contentFromRemoteConfig)
    }
  }

  override fun sendUpdateInfoDisplayed()
  {

  }

  override fun sendRecomendedUpdateDisplayed()
  {

  }

  override fun sendBlockingUpdateDisplayed()
  {

  }

  override fun sendAskLaterEvent()
  {

  }

  override fun closeAppWithoutUpdateEvent()
  {

  }

  override fun sendInformativeActionButtonEvent()
  {

  }

  override fun sendRecommendedActionButtonEvent()
  {

  }

  override fun sendBlockingActionButtonEvent()
  {

  }

  override fun getVersionCode(): Int
  {
    return 0
  }

  override fun getDateForAnalytics(): String?
  {
    return null
  }

  override fun generateAnalyticsExtraInfos(): Bundle?
  {
    return null
  }
}
